main.cpp

```diff
@@ -1 +1,116 @@
+#include <iostream>
+#include <vector>
+#include <algorithm>
 
+const std::vector<std::vector<int>> matrix{{
+                                                   { 1 , 8 , 7 , 19, 3 , 7 , 5 , 11, 5 , 4 },
+                                                   { 1 , 3 , 2 , 1 , 13, 6 , 25, 8 , 9 , 15},
+                                                   { 1 , 1 , 1 , 4 , 6 , 3 , 17, 6 , 10, 13},
+                                                   { 13, 4 , 1 , 1 , 9 , 7 , 12, 11, 5 , 15},
+                                                   { 17, 12, 8 , 1 , 3 , 1 , 1 , 1 , 10, 4 },
+                                                   { 7 , 4 , 19, 1 , 11, 1 , 6 , 1 , 2 , 7 },
+                                                   { 9 , 10, 12, 1 , 1 , 1 , 13, 1 , 17, 5 },
+                                                   { 3 , 5 , 9 , 14, 10, 18, 5 , 1 , 3 , 10},
+                                                   { 14, 2 , 15, 8 , 19, 17, 4 , 1 , 5 , 4 },
+                                                   { 13, 8 , 6 , 3 , 3 , 7 , 5 , 1 , 1 , 1 }
+                                            }};
+
+struct node{
+    int key;
+    int i;
+    int j;
+    bool visited;
+    int mass;
+};
+
+
+std::vector<std::vector<node>> InitGraph(std::vector<std::vector<int>> m){
+    std::vector<std::vector<node>> Graph;
+    auto N = m.size();
+    for(auto i=0;i<N;i++){
+        Graph.emplace_back(std::vector<node>{m[i].size()});
+        for(auto j=0;j<m[i].size();j++){
+            Graph[i][j] =node{m[i][j], i, j, false, 100000};
+        }
+    }
```

Comment:
Codestyle. Его нет. Рекомендуется взять какой-то `.clang-format` от гугла или llvm и скормить его IDE. После чего периодически протикивать reformat для всего файла, а в идеале поставить reformat on save  в настройках IDE.

Reply:
Не стоит игнорировать `range-based` циклы. 
```suggestion
    for (const auto &input_line: m){
        Graph.emplace_back(std::vector<node>{input_line.size()});
        for (auto value : input_line) {
            Graph[i][j] = node{value, i, j, false, 100000};
        }
    }
```